docs/elasticsearch/ux.md

```diff
@@ -0,0 +1,1240 @@
+
+To implement a "Did you mean `suggestion`?" feature, use a `phrase` suggestor.
+This is similar to the `term` suggestor, except that it uses N-gram language models to suggest whole phrases instead of individual words.
+
+Create a custom analyzer called `trigram` that uses a `shingle` filter. This filter is the same as the `edge_ngram` filter, except that it applies to words instead of letters.
+
+```json
+PUT shakespeare
+{
+  "settings": {
+    "index": {
+      "analysis": {
+        "analyzer": {
+          "trigram": {
+            "type": "custom",
+            "tokenizer": "standard",
+            "filter": [
+              "lowercase",
+              "shingle"
+            ]
+          }
+        },
+        "filter": {
+          "shingle": {
+            "type": "shingle",
+            "min_shingle_size": 2,
+            "max_shingle_size": 3
+          }
+        }
+      }
+    }
+  },
+  "mappings": {
+    "properties": {
+      "text_entry": {
+        "type": "text",
+        "fields": {
+          "trigram": {
+            "type": "text",
+            "analyzer": "trigram"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+If you run a query with an incorrect phrase:
+
+```json
+POST shakespeare/_search
+{
+  "suggest": {
+    "text": "That is the qution",
+    "simple_phrase": {
+      "phrase": {
+        "field": "text_entry.trigram"
+      }
+    }
+  }
+}
+```
+
+You get back the corrected phrase:
+
+```json
+{
+  "took": 3,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 0,
+      "relation": "eq"
+    },
+    "max_score": null,
+    "hits": []
+  },
+  "suggest": {
+    "simple_phrase": [
+      {
+        "text": "That is the qution",
+        "offset": 0,
+        "length": 18,
+        "options": [
+          {
+            "text": "that is the question",
+            "score": 0.0015543294
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+
+## Pagination
+
+Use the `from` and `size` parameters to return results to your users one page at time.
+
+The `from` parameter indicates the document number that you want to start showing the results from. The `size` parameter refers to the number of results that you want to show. The `from` and `size` parameters together act as a sliding window that moves through the results.
+
+For example, if the value of `size` is 10 and the value of `from` is 0, you see the first 10 results. If you change the value of `from` to 10, you see the next 10 results (because the results are zero-indexed). So, if you want to see results starting from result 11, `from` must be 10.
+
+```json
+GET shakespeare/_search
+{
+  "from": 0,
+  "size": 10,
+  "query": {
+    "match": {
+      "play_name": "Hamlet"
+    }
+  }
+}
+```
+
+```json
+Search Results
+------------- from: 0, size: 10
```

Comment:
I see what you're doing here, but I think you should probably stick with the zero-indexing and say 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 | 10, 11, 12, etc.

Reply:
Yes, it does take up too much space. I think saying "zero indexed" and "if you want to see results starting from result 11, `from` must be 10" explains it well enough.